File: find-k-pairs-with-smallest-sums/find-k-pairs-with-smallest-sums.java
class Solution {
    public List<List<Integer>> kSmallestPairs(int[] nums1, int[] nums2, int k) {
        var pq = new PriorityQueue<List<Integer>>((a,b) -> (b.get(0)+b.get(1)) - (a.get(0)+a.get(1)));
        for(int i: nums1) {
            for (int j: nums2) {
                pq.add(List.of(i, j));
                if (pq.size() > k) 
                    pq.poll();
            }
        }
        var tempList = new ArrayList<List<Integer>>();
        while(pq.size() != 0) {
            tempList.add(pq.poll());
        }
        return tempList;
    }
    
    public void print(Object o) {
        System.out.println(o);
    }
}

File: find-k-pairs-with-smallest-sums/find-k-pairs-with-smallest-sums.py
class Solution:
    def kSmallestPairs(self, nums1: List[int], nums2: List[int], k: int) -> List[List[int]]:
        heap = []
        
        for i in range(len(nums1)):
            for j in range(len(nums2)):
                heappush(heap, (-1 * (nums1[i]+nums2[j]), (nums1[i], nums2[j])))
                if len(heap) > k:
                    heappop(heap)
        
        ret = []
        for _ in range(len(heap)):
            ret.append(heappop(heap)[1])
        return ret